lib/src/frontend/spawn_hybrid.dart

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:async/async.dart';
+import 'package:path/path.dart' as p;
+import 'package:stream_channel/stream_channel.dart';
+
+import '../backend/invoker.dart';
+import '../util/remote_exception.dart';
+import '../utils.dart';
+
+/// A transformer that handles messages from the spawned isolate and ensures
+/// that messages sent to it are JSON-encodable.
+final _transformer = new StreamChannelTransformer(
+    new StreamTransformer.fromHandlers(handleData: (message, sink) {
+      switch (message['type']) {
+        case 'data':
+          sink.add(message['data']);
+          break;
+
+        case 'print':
+          print(message['line']);
+          break;
+
+        case 'error':
+          var error = RemoteException.deserialize(message['error']);
+          sink.addError(error.error, error.stackTrace);
+          break;
+      }
+    }), new StreamSinkTransformer.fromHandlers(handleData: (message, sink) {
+      ensureJsonEncodable(message);
+      sink.add(message);
+    }));
+
+/// Spawns a VM isolate for the given [uri], which may be a [Uri] or a [String].
+///
+/// This allows browser tests to spawn servers with which they can communicate
+/// to test client/server interactions. It can also be used by VM tests to
+/// easily spawn an isolate.
+///
+/// The Dart file at [uri] must define a top-level `hybridMain()` function that
+/// takes a `StreamChannel` argument and, optionally, an `Object` argument to
```

Comment:
You can, but the Stream API doesn't make it particularly easy to get the first event without also canceling the stream subscription. This also mirrors the Isolate spawn APIs more closely.